plugins/modules/system/puppet.py

```diff
@@ -255,6 +260,8 @@ def main():
             cmd += " --server %s" % shlex_quote(p['puppetmaster'])
         if p['show_diff']:
             cmd += " --show_diff"
+        if p['confdir']:
+            cmd += " --confdir '%s'" % p['confdir']
```

Comment:
Better use
```suggestion
            cmd += " --confdir %s" % shlex_quote(p['confdir'])
```
to correctly handle quoting.

(It would make sense to change the module to use a list of strings instead of a string for `cmd`, then quoting wouldn't be necessary at all.)